src/core/appdir.cpp

```diff
@@ -834,7 +834,15 @@ namespace linuxdeploy {
                     if (!d->deployElfDependencies(sharedLibrary))
                         return false;
 
-                    d->setElfRPathOperations[sharedLibrary] = "$ORIGIN";
+                    const auto rpath = elf_file::ElfFile(sharedLibrary).getRPath();
+                    auto rpathList = util::split(rpath, ':');
+                    if (std::find(rpathList.begin(), rpathList.end(), "$ORIGIN") == rpathList.end())
+                    {
```

Comment:
```suggestion
                    if (std::find(rpathList.begin(), rpathList.end(), "$ORIGIN") == rpathList.end()) {
```
Code style: `{` must be on the same line the `if` is on.